beacon_node/network/src/network_beacon_processor/rpc_methods.rs

```diff
@@ -815,16 +815,36 @@ impl<T: BeaconChainTypes> NetworkBeaconProcessor<T> {
             .start_slot(T::EthSpec::slots_per_epoch());
 
         let (block_roots, block_roots_source) = if req_start_slot >= finalized_slot.as_u64() {
+            // If the entire requested range is after finalization, use fork_choice
             (
                 self.chain
                     .block_roots_from_fork_choice(req_start_slot, req_count),
                 "fork_choice",
             )
-        } else {
+        } else if req_start_slot + req_count <= finalized_slot.as_u64() + 1 {
+            // If the entire requested range is before finalization, use store
             (
                 self.get_block_roots_from_store(req_start_slot, req_count)?,
                 "store",
             )
+        } else {
+            // Split the request at the finalization boundary
+            let count_from_store = finalized_slot.as_u64() - req_start_slot + 1;
+            let count_from_fork_choice = req_count - count_from_store;
+            let start_slot_fork_choice = finalized_slot.as_u64() + 1;
```

Comment:
```suggestion
            let start_slot_fork_choice = finalized_slot.as_u64();
```

Want to fetch all slots `>= finalized_slot` from fork choice.